pkg/trace/agent/agent.go

```diff
@@ -72,6 +73,7 @@ func NewAgent(ctx context.Context, conf *config.AgentConfig) *Agent {
 	agnt := &Agent{
 		Concentrator:       stats.NewConcentrator(conf.BucketInterval.Nanoseconds(), statsChan),
 		Blacklister:        filters.NewBlacklister(conf.Ignore["resource"]),
+		Validator:          filters.NewValidator(conf.RequiredTags),
```

Comment:
+1 looks good.

We need to make sure `conf.RequiredTags` gets populated from the config file/environment. See how `conf.Ignore["resource"]` is set.